diff --git a/paragraphs.install b/paragraphs.install
index a806227..3fcaedd 100644
--- a/paragraphs.install
+++ b/paragraphs.install
@@ -126,6 +126,24 @@ function paragraphs_field_schema($field) {
 }
 
 /**
+ * Implements hook_install().
+ */
+function paragraphs_install() {
+  paragraphs_ensure_entitycache_table();
+}
+
+/**
+ * Implements hook_uninstall().
+ */
+function paragraphs_uninstall() {
+  // We can't use paragraphs_remove_entitycache_table() here.
+  // The module isn't loaded here.
+  if (db_table_exists('cache_entity_paragraphs_item')) {
+    db_drop_table('cache_entity_paragraphs_item');
+  }
+}
+
+/**
  * Make sure all paragraph fields have the new index on revision_id.
  */
 function paragraphs_update_7100() {
@@ -136,13 +154,14 @@ function paragraphs_update_7100() {
 }
 
 /**
+ * Deprecated. Functionality is now in 7102.
+ */
+function paragraphs_update_7101() { }
+
+/**
  * Make sure the entitycache table exists.
  */
-function paragraphs_update_7101() {
-  if (module_exists('entitycache') && !db_table_exists("cache_entity_paragraphs_item")) {
-    drupal_load('module', 'entitycache');
-    $cache_schema = drupal_get_schema_unprocessed('system', 'cache');
-    $cache_schema['description'] = "Cache table used to store paragraphs_item entity records.";
-    db_create_table("cache_entity_paragraphs_item", $cache_schema);
-  }
+function paragraphs_update_7102() {
+  module_load_include('module', 'paragraphs');
+  paragraphs_ensure_entitycache_table();
 }
diff --git a/paragraphs.module b/paragraphs.module
index 973b785..e1c763e 100644
--- a/paragraphs.module
+++ b/paragraphs.module
@@ -1366,3 +1366,42 @@ function paragraphs_bundle_copy_info() {
     ),
   );
 }
+
+/**
+ * Helper to ensure entitycache table.
+ */
+function paragraphs_ensure_entitycache_table() {
+  if (module_exists('entitycache') && !db_table_exists('cache_entity_paragraphs_item')) {
+    drupal_load('module', 'entitycache');
+    $cache_schema = drupal_get_schema_unprocessed('system', 'cache');
+    $cache_schema['description'] = 'Cache table used to store paragraphs_item entity records.';
+    db_create_table('cache_entity_paragraphs_item', $cache_schema);
+  }
+}
+
+/**
+ * Helper to remove entitycache table.
+ */
+function paragraphs_remove_entitycache_table() {
+  if (db_table_exists('cache_entity_paragraphs_item')) {
+    db_drop_table('cache_entity_paragraphs_item');
+  }
+}
+
+/**
+ * Implements hook_modules_installed().
+ */
+function paragraphs_modules_installed($modules) {
+  if (in_array('entitycache', $modules)) {
+    paragraphs_ensure_entitycache_table();
+  }
+}
+
+/**
+ * Implements hook_modules_uninstalled().
+ */
+function paragraphs_modules_uninstalled($modules) {
+  if (in_array('entitycache', $modules)) {
+    paragraphs_remove_entitycache_table();
+  }
+}
\ No newline at end of file
